### PR TITLE
Edit profile page has recurring contribution tab

### PR DESCRIPTION
**Pull request by JustinPinner**
## What does this change?
Adds a switch to enable/disable the appearance of a Contributions tab on the edit profile page. When enabled, recurring contributors can update their card details in the event of expired or declined cards.

## What is the value of this and can you measure success?
We're currently missing the feature to enable recurring contributors to maintain their own card details, and so they are dependent on calling in to user help.

## Does this affect other platforms - Amp, Apps, etc?
No, web only.

## Does this affect GLabs Paid Content Pages? Should it have support for Paid Content?
No.

## Screenshots
**Switched ON**
screen shot 2017-10-31 at 10 51 00

**Change card**
screen shot 2017-10-31 at 10 51 14

screen shot 2017-10-31 at 10 52 17

screen shot 2017-10-31 at 10 52 28

screen shot 2017-10-31 at 10 52 37

**Not a known contributor**
screen shot 2017-10-30 at 17 49 28
 

**Switched OFF**
screen shot 2017-10-27 at 15 39 10

## Tested in CODE?
YES

### CODE REVIEW

File: static/src/javascripts/projects/membership/contributions-recurring-tab.js
Comment (jacobwinch) [inline static/src/javascripts/projects/membership/contributions-recurring-tab.js]: Personally I would prefer to move away from exposing subscription names/numbers to users - but I guess UX will make the final call on that!

File: static/src/javascripts/projects/membership/contributions-recurring-tab.js
Comment (jacobwinch) [inline static/src/javascripts/projects/membership/contributions-recurring-tab.js]: This is probably on your to-do list already, but we should make sure to remove all of these membership related references/comments before merging this.

File: static/src/javascripts/projects/membership/contributions-recurring-tab.js
Comment (jacobwinch) [reply static/src/javascripts/projects/membership/contributions-recurring-tab.js]: In terms of an internal reference number, the users email address or identity id should be sufficient for userhelp to identify their regular contribution in Zuora

**Review by Ap0c — APPROVED** [review]
🥐 Ran it locally and it worked!

File: static/src/javascripts/projects/membership/contributions-recurring-tab.js
Comment (JustinPinner) [reply static/src/javascripts/projects/membership/contributions-recurring-tab.js]: Agreed. Elvis and I had a conversation with user help yesterday and arrived at the same conclusion, so we've removed the Contributor ref :)

File: static/src/javascripts/projects/membership/contributions-recurring-tab.js
Comment (JustinPinner) [reply static/src/javascripts/projects/membership/contributions-recurring-tab.js]: I _think_ they're all gone now. Will re-check prior to merging anyway.

**Review by mariogalic — APPROVED** [review]

**Pull request merged**
